static/js/display.js

```diff
@@ -0,0 +1,213 @@
+    var start = (page-1)*perPage
+    var end = page*perPage
+    for(let obj of imgs.slice(start,end)){
+        var row = img_table.insertRow();
+        for (let type of data_types){
+            if(type[0] === obj['type_id']){
+                var target_type = type
+            }
+        }
+        var preview_img = document.createElement('img');
+        preview_img.setAttribute('src',obj['img_url']);
+        preview_img.setAttribute('width','240');
+        preview_img.setAttribute('height','180');
+
+        var editButton = create_editButton(type='image',obj['img_id']);
+        var deleteButton = create_deleteButton(obj['img_id']);
+
+        row.insertCell().innerHTML = obj['img_id'];
+        row.insertCell().innerHTML = target_type[1];
+        row.insertCell().innerHTML = obj['img_upload_time'];
+        row.insertCell().innerHTML = obj['img_start_date'];
+        row.insertCell().innerHTML = obj['img_end_date'];
+        row.insertCell().innerHTML = obj['img_start_time'];
+        row.insertCell().innerHTML = obj['img_end_time'];
+        row.insertCell().innerHTML = obj['img_display_count'];
+        row.insertCell().innerHTML = obj['img_display_time'];
+        row.insertCell().appendChild(preview_img);
+        var editCell = row.insertCell();
+        editCell.appendChild(editButton);
+        editCell.appendChild(deleteButton);
+    }
+    var pageList=[]
+    if(Number(page)>=3){
+        if(Number(page)<=totalPage-2){
+            pageList.push(Number(page)-2,Number(page)-1,Number(page),Number(page)+1,Number(page)+2);
+        }
+        else if(Number(page)===totalPage-1){
+            pageList.push(Number(page)-3,Number(page)-2,Number(page)-1,Number(page),Number(page)+1);
+        }
+        else{
+            pageList.push(Number(page)-4,Number(page)-3,Number(page)-2,Number(page)-1,Number(page));
+        }
+    }else{
+        if(totalPage<=5){
+            for(let i=page;i<=totalPage;i++){
+                pageList.push(i);
+            }
+        }else{
+            for(let i=page;i<=5;i++){
+                pageList.push(i);
+            }
+        }
+    }
+    create_pagination(pageList,type);
+}
+
+function insert_texts(texts,data_types,page,type){
+    text_table = document.getElementById('text_table');
+    while(text_table.rows.length > 1){
+        text_table.deleteRow(-1);
+    }
+    var count = texts.length
+    var perPage = 8;
+    var totalPage = Math.ceil(count/perPage)
+    var start = (page-1)*perPage
+    var end = page*perPage
+    for(let text of texts.slice(start,end)){
+        var row = text_table.insertRow();
+        for (let type of data_types){
+            if(type[0] === text[1]){
+                var target_type = type
+            }
+        }
+        
+        var editButton = create_editButton(type='text',text[0]);
+        var deleteButton = create_deleteButton(text[0]);
+
+        row.insertCell().innerHTML = text[0];
+        row.insertCell().innerHTML = target_type[1];
```

Comment:
What if `type[0] !== text[1]`, will the variable exist?

Reply:
OK, I will add check. 
When the data_type doesn't exists I will let this field to be empty string.

Reply:
I think `type[0] !== text[1]` hardly happen. Because when uploading a text, user needs to select the data_type which has already existed in DB. So, `type[0] !== text[1]` maybe only occurs when this data type delete from DB. If it really happened, I guess it may cause error due to variable undefined.

Reply:
Fixed~